pkg/storaged/resize.jsx

```diff
@@ -0,0 +1,365 @@
+
+            if (!info) {
+                info = { };
+                shrink_excuse = grow_excuse = cockpit.format(_("$0 filesystems can not be resized here."),
+                                                             block.IdType);
+            } else {
+                if (!info.can_shrink)
+                    shrink_excuse = cockpit.format(_("$0 filesystems can not be made smaller."),
+                                                   block.IdType);
+                if (!info.can_grow)
+                    grow_excuse = cockpit.format(_("$0 filesystems can not be made larger."),
+                                                 block.IdType);
+            }
+        } else if (client.blocks_stratis_blockdev[block.path] && client.features.stratis_grow_blockdevs) {
+            info = {
+                can_shrink: false,
+                can_grow: true,
+                grow_needs_unmount: false
+            };
+            shrink_excuse = _("Stratis blockdevs can not be made smaller");
+        } else if (block.IdUsage == 'raid') {
+            info = { };
+            shrink_excuse = grow_excuse = _("Physical volumes can not be resized here.");
+        } else if (client.legacy_vdo_overlay.find_by_backing_block(block)) {
+            info = {
+                can_shrink: false,
+                can_grow: true,
+                grow_needs_unmount: false
+            };
+            shrink_excuse = _("VDO backing devices can not be made smaller");
+        } else {
+            info = {
+                can_shrink: false,
+                can_grow: true,
+                grow_needs_unmount: true
+            };
+            shrink_excuse = _("Unrecognized data can not be made smaller here.");
+        }
+        if (to_fit) {
+            // Shrink to fit doesn't need to resize the content
+            shrink_excuse = null;
+        }
+    } else {
+        info = { };
+        shrink_excuse = grow_excuse = _("This volume needs to be activated before it can be resized.");
+    }
+
+    return { info, shrink_excuse, grow_excuse };
+}
+
+export function grow_dialog(client, lvol, info, to_fit) {
+    const block = client.lvols_block[lvol.path];
+    const vgroup = client.vgroups[lvol.VolumeGroup];
+    const pool = client.lvols[lvol.ThinPool];
+
+    const usage = get_active_usage(client, block && info.grow_needs_unmount ? block.path : null, _("grow"));
+
+    if (usage.Blocking) {
+        dialog_open({
+            Title: cockpit.format(_("$0 is in use"), lvol.Name),
+            Body: BlockingMessage(usage)
+        });
+        return;
+    }
+
+    let grow_size;
+    let size_fields = [];
+    if (!to_fit) {
+        size_fields = [
+            SizeSlider("size", _("Size"),
+                       {
+                           value: lvol.Size,
+                           min: lvol.Size,
+                           max: (pool ? pool.Size * 3 : lvol.Size + vgroup.FreeSize),
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19400-20230927-134545-de938f8a-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)